index.js

```diff
@@ -196,10 +197,23 @@ module.exports = {
         groups: { rendering: ["/^render.+$/", "render"] }
       }
     ],
+    "react/no-redundant-should-component-update": "error",
+    "react/no-will-update-set-state": "error",
+    "react/no-unsafe": "error",
+    "react/void-dom-elements-no-children": "error",
+    "react/jsx-sort-props": "error",
```

Comment:
Quick question: do you feel it still worth enforcing it ?
We (mostly me :-)) kinda think that alphabetical order brings more confusion than having a functional grouping of props (either in JSX or in interface). Also, we're not consistent since we enforce it in jsx but not in interface for example. Do you SC guys still make those kind of comments on PR ?

Reply:
I don't know if it's worth enforcing or not, we didn't have to fix a lot of them in SC codebase so I figured we might as well activate it. But it's been quite some time since I've seen a comment about this in PR reviews, it doesn't happen much.

The main advantage I see to the alphabetical order is:
* No need to wonder where to put a prop and try to figure out the existing logical/functional grouping

Also it can:
* Avoid some useless merge conflicts
* And if the rule is enforced like this, no need to have a look at that stuff in manual reviews anymore

If it doesn't make sense for SQ codebase anymore, we can disable it here (or keep as warn) and only enforce it on our side...

Reply:
I think that moving that rule from `warn` to `error`doesn't change much, since it was already reported as an error during jest tests (aka fixed anyway if tested).

The main thing I'm challenging here is: do you still make those kinds (alphabetical) of comments on your PRs ? The answer is yes.

We SQ team agreed that it would make more sense to group them by functional/technical area (eg: loading & disabled & ... together, callbacks together, functionally related props together). All that to ease the reading of the code. It is difficult to gather all relevant props when they're spread across the interface.

<hr/>

> No need to wonder where to put a prop and try to figure out the existing logical/functional grouping

I believe this is introducing some sort of maintainability debt. It is much easier for someone to figure out where to put props while adding them (because you need to have a certain level of awareness at that time) than leaving this responsibility to the one that will read it later. WDYT ?

> Avoid some useless merge conflicts

Do you recall suffering from that in the past ?

Reply:
> Do you recall suffering from that in the past ?

Not at all thanks to this rule! :stuck_out_tongue_winking_eye: Nah but I think it's a fake argument, it's not like we often work on the same components anyway. And if it happen, there will be conflict somewhere else.

> WDYT ?

I agree with the reasons why you dropped this rule in SQ team, it's probably often easier to read and understand the code when those are *correctly* grouped in a logical way. But at the same time I'm a bit afraid that it could also slip very easily into a mess, with bad groupings that are logical only to the one who wrote them, or some devs ignoring/missing current logical order. And you end up having to challenge that in PR reviews.

So for now I think we will keep our current way of doing on SC side (unless someone else from SC team has a strong opinion about this?). I'll remove the rule from this common config and only enforce it on our side.